### ISSUE TITLE
Merge android-6.0.1_r79 (MOB31T)

### ISSUE THREAD

**Issue by pdsouza** [maruos/manifest#10]
Tags: https://groups.google.com/d/msg/android-building/racl35q1MCk/ql0uesE4AwAJ

Repositories to merge:

## Platform

- [x] android_platform_frameworks_native (https://github.com/maruos/android_platform_frameworks_native/commit/31ff97c131b01e1a4653ae0b0ed6221fac6b6ff8)
- [x] android_platform_frameworks_base (https://github.com/maruos/android_platform_frameworks_base/commit/158e5032a4d1bd1c14f0e698a64e8487df34c497)
- [x] android_platform_hardware_qcom_display (no changes)
- [x] android_platform_packages_apps_Settings (no changes)
- [x] android_platform_build (https://github.com/maruos/android_platform_build/commit/dd2bbbb9fe9cb7d1653902a0e10a0037cad55ac0)

## Device

- [x] android_device_lge_hammerhead (no changes)
- [x] android_device_asus_flo (no changes)

## Kernel

- [x] android_kernel_hammerhead (no changes...past support period for hammerhead)
- [x] android_kernel_flo (no changes...past support period for flo)

## Manifest

- [x] manifest (https://github.com/maruos/manifest/commit/12cf02cd0a830d5f4f3dc3d165277f93161e2c79)

*NOTE: This is on `marhsmallow-mr2-release` so we need to cherry these in vs. a merge since master is based on `marshmallow-mr3-release`; same deal as MOB31S.*

**Comment by pdsouza** [comment]
```
frameworks/native $ git cherry -v maruos/master android-6.0.1_r79
- 3bcf0caa8cca9143443814b36676b3bae33a4368 Add FrameStats default constructor
- 1ecb999624165ea2a9cf5e16b3f7e93358d503d6 Region: Detect malicious overflow in unflatten
- ebf0c917c3034002b3ca8abb22f7417b6ecc7e30 DO NOT MERGE ServiceManager: Restore basic uid check
- 23e7c1d038bae581cf896f72706221c2ef610c61 ServiceManager: Allow system services running as secondary users to add services
- b5c8da661e8c6cd32d038d4fd9b5bc2f6f4046c4 Fix SF security vulnerability: 32660278
+ 80fef3303ad34be1f6fefc7adfd84632f5852eb9 Correct overflow check in Parcel resize code
+ b4ad8fd532d4a4dcc4329760b88e92ca0cf50181 Fix SF security vulnerability: 32706020
```

2 commits from frameworks/native.

**Comment by pdsouza** [comment]
```
frameworks/base $ git cherry -v maruos/master android-6.0.1_r79
- e7cf91a198de995c7440b3b64352effd2e309906 Don't trust callers to supply app info to bindBackupAgent()
- a75537b496e9df71c74c1d045ba5569631a16298 DO NOT MERGE Fix intent filter priorities
- 468651c86a8adb7aa56c708d2348e99022088af3 DO NOT MERGE Disable app pinning when emergency call button pressed
- 01875b0274e74f97edf6b0d5c92de822e0555d03 Reduce shell power over user management.
- 4e4743a354e26467318b437892a9980eb9b8328a Add pm operation to set user restrictions.
- 55271d454881b67ff38485fdd97598c542cc2d55 WifiEnterpriseConfiguration: Do not print credentials in toString
- 81be4e3aac55305cbb5c9d523cf5c96c66604b39 Fix string equality comparison
- 28460c294ebd9c9cab258c92a20e0e2cc63e64f0 Check caller's uid before allowing notification policy access.
- f0ea4c8737ec911074cc4d284f984c5c353f32e2 Add bound checks to utf16_to_utf8
- 69729fa8b13cadbf3173fe1f389fe4f3b7bd0f9c DO NOT MERGE Block the user from entering safe boot mode
- e739d9ca5469ed30129d0fa228e3d0f2878671ac DO NOT MERGE: Remove the use of JHEAD in ExifInterface
- 5aa2601ddd7c52b740cdffdeb8afd22d0151b1d5 DO NOT MERGE: Fix CTS regression
- 90ea10c4d3e4aceba98276f5f16389aa6f99584a DO NOT MERGE: Allow apps with CREATE_USERS permission to call UM.getProfiles.
- c998deb9376eea9289d81ac03b41e01722c8a761 DO NOT MERGE: Clean up when recycling a pid with a pending launch
- 1d6c0efc202a21942321ffa3b83b7c9309e66c9a Fix vulnerability in LockSettings service
- a1e18814d23f36c0ec970d6e04de3dbc1214d53d Process: Fix communication with zygote.
- ca692c228dfef0b0d7f51597e726180d3f70c66c Bind fingerprint when we start authentication - DO NOT MERGE
- f1e317003659e9097f760a4b680dd595abed2e3e DO NOT MERGE: Fix vulnerability where large GPS XTRA data can be injected. -Can potentially crash system with OOM. Bug: 29555864
- b8be33b0bedec211708c4525b9d3f3b4effb385c Backport changes to whitelist sockets opened by the zygote.
- 23e6629cf56378634488ea49a8acf40d709effa1 DO NOT MERGE) ExifInterface: Make saveAttributes throw an exception before change
- 26f6eb78a61323cb2cae055ee0cee62c95a8f85e Fix setPairingConfirmation permissions issue (2/2)
- 77fa5d963f796f7682fd3f859e3f148681f60bde Ensure munmap matches mmap
- 115baf8f0fb54fa86db6afc489ab8413ff714e46 DO NOT MERGE: Fix deadlock in AcitivityManagerService.
- 9e812be46b6f7be61524e14652ba380c02518262 DO NOT MERGE: Catch all exceptions when parsing IME meta data
- 4f0dec2e62ee47dec1e29a6c97e13af2c86a33e9 Fix build break due to automerge of 7d2198b5
- 4c95059c31584cbd5651183a4605301563cf2e8e DO NOT MERGE Check caller for sending media key to global priority session
- f653d365122f0cebc92ac7bb0af805cc4d80f638 Merge commit '0a9d537f623b2c11dce707fb9b91fea016fd0e9f' into manual_merge_0a9d537
- ef525e8d721a69310044eca14993df8b13ac1b14 Avoid crashing when downloading MitM'd PAC that is too big am: 7d2198b586 am: 9c1cb7a273 am: 6634e90ad7 am: 66ee2296a9
- 248c3ab8323c774d213df75ac528d4ab96107fe4 DO NOT MERGE) ExifInterface: Provide backward compatibility
- 37648d47981bf9a52738dbe534982e0bdfd98eff DO NOT MERGE Isolated processes don't get precached system service binders
- 81a958fc287bc8b1c0601960f466828f4abb0c2c DO NOT MERGE) ExifInterface: Close the file when an exception happens
- cc3a845d08fc99b1af1a1707492495a34cb2a1cb Zygote : Block SIGCHLD during fork.
- 80b629288444d7841dd7a3e0daf46c263c818445 Public volumes belong to a single user.
- 7aa8ec22a3d4a834007d0e6f57a078f773dd729b Zygote: Additional whitelists for runtime overlay / other static resources.
- c3ee76203a1ce9b8c490975574707c9389c77057 Zygote: Additional whitelisting for legacy devices.
- 94946899615c2af2eaa39895a1242499d078c3c8 Fix idmap leak in zygote process
- db66afe61e52f81f185593a2b6dd63c436ce9014 resolve merge conflicts of 89aa6fb to mnc-dr-dev
+ cefaefa4f447a51a717e7cca750461d3f19b68fc DO NOT MERGE: Check provider access for content changes.
+ 12033ddaf11ae11d0212eabcf82639bd73410e47 DO NOT MERGE. Retain DownloadManager Uri grants when clearing.
```

2 commits from frameworks/base.

**Comment by pdsouza** [comment]
For reference, a summary of all changes:
```
workspace $ repo forall -p -c git log --oneline android-6.0.1_r78..android-6.0.1_r79

project bionic/
e7722dd Check for bad packets in getaddrinfo.c's getanswer.

project build/
bbaab61 MOB31T
c19639c [DO NOT MERGE] Update Security String to 2017-02-01 on mnc-dev

project device/htc/flounder/
9184e75 Fix security issue in Visualizer effect

project external/libavc/
30ea86c Decoder: Fix in checking for valid profile flags

project external/libhevc/
3da3ec6 Fixed handling invalid chroma tu size for error clips
5e7a614 Fixed out of bound reads in stack variables
02bcb7d Fix in Chroma SAO for non-multiple of 8 height

project frameworks/av/
8cc6882 Fix security vulnerability: potential OOB write in audioserver
e537af7 Effect: Use local cached data for Effect commit

project frameworks/base/
12033dd DO NOT MERGE. Retain DownloadManager Uri grants when clearing.
cefaefa DO NOT MERGE: Check provider access for content changes.

project frameworks/native/
b4ad8fd Fix SF security vulnerability: 32706020
80fef33 Correct overflow check in Parcel resize code

project hardware/libhardware/
740a332 Fix security vulnerability: potential OOB write in audioserver

project packages/apps/Bluetooth/
1d44ec7 Remove MANAGE_DOCUMENTS permission as it isn't needed

project packages/apps/Messaging/
e1db4da 32807795  Security Vulnerability - AOSP Messaging App: thirdparty can attach private files from "/data/data/com.android.messaging/" directory to the m
9aaf452 32161610 Security Vulnerability - Information disclosure vulnerability in AOSP Messaging
07f7fb9 32322450 Security Vulnerability - heap buffer overflow in libgiftranscode.so

project packages/apps/UnifiedEmail/
e1ba54e Don't allow file attachment from /data through GET_CONTENT.
```
